### PR TITLE
Add New-MiLink cmdlet

### PR DESCRIPTION
**Pull request by ostojicb**
this cmdlet can be ran on an already set-up environment (AlwaysOn, networking to Mi etc) - i.e.  should be more or less scoped to SSMS wizard functionality. 

there will be at least one more cmdlet for env setup, and we might add few more steps (or checks at least) in this cmdlet

### CODE REVIEW

File: cmdlets/sqlserver/New-MiLink.ps1 [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        # TODO: If we don't have DBM endpoint, we have to create it
+        # we can either create a new certificate for that purpose, or pick an existing one if it meets certain conditions such as:
+        # - "select pvt_key_encryption_type from sys.certificates where name = N'$boxCertName'" must be MK
+        # TODO: can we trust sp_get_endpoint_certificate ?
+        Write-Verbose "Validate DBM endpoint certificate [started]"
+        $dbmCert = Invoke-Sqlcmd -Query "exec sp_get_endpoint_certificate @endpoint_type = 4" -ServerInstance $SqlInstance -OutputSqlErrors:$false
+        if (!$dbmCert) {
+            $boxCertName = Read-Host "Creating a new certificate for DBM Endpoint. Enter cert name: "
+            $boxCertSubject = Read-Host "Enter cert subject: "
+            $queryCreateCertificate = "CREATE CERTIFICATE $boxCertName WITH SUBJECT = N'$boxCertSubject', EXPIRY_DATE = N'12/12/2030';"
```
Comment (bruzel) [inline cmdlets/sqlserver/New-MiLink.ps1:169]: Maybe you should not hard-code this date- better use getdate().AddYears(10) or something similar.

File: cmdlets/sqlserver/New-MiLink.ps1 [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        Write-Verbose "Export SQL server DBM endpoint certificate to MI [completed]"
+
+        #
+        Write-Verbose "Create availability groups [started]"
+        $queryAG =
+        @"
+CREATE AVAILABILITY GROUP [$PrimaryAvailabilityGroup]
+WITH (CLUSTER_TYPE = NONE)
+FOR DATABASE [$DatabaseName]
+REPLICA ON N'chimera' WITH (ENDPOINT_URL = N'$sourceEndpoint', FAILOVER_MODE = MANUAL, AVAILABILITY_MODE = SYNCHRONOUS_COMMIT, SEEDING_MODE = AUTOMATIC, SECONDARY_ROLE(ALLOW_CONNECTIONS = ALL));
```
Comment (bruzel) [inline cmdlets/sqlserver/New-MiLink.ps1:233]: What is 'chimera'?

Comment (ostojicb) [reply cmdlets/sqlserver/New-MiLink.ps1:233]: good catch, that should be `$SqlInstance`

File: cmdlets/sqlserver/New-MiLink.ps1 [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+            Get-SqlBackupHistory -ServerInstance $SqlInstance -DatabaseName $DatabaseName
+        }
+
+        Write-Verbose "Get MI info ($ManagedInstanceName) [started]"
+        $instance = Get-AzSqlInstance -ResourceGroupName $ResourceGroupName -Name $ManagedInstanceName
+        $miFQDN = $instance.FullyQualifiedDomainName       
+        $SqlMiInstanceConnectionString = "tcp://${miFQDN}:5022;Server=[$ManagedInstanceName]"
+        Write-Verbose "Get MI info ($ManagedInstanceName) [completed]"
+
+        # Figure out if we're on a VM https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=windows#example-scenarios-for-usage
```
Comment (bruzel) [inline cmdlets/sqlserver/New-MiLink.ps1:142]: I think you just need to fail if you detect we're not on the VM.

Comment (ostojicb) [reply cmdlets/sqlserver/New-MiLink.ps1:142]: i agree. I think I will add a `begin { }` block that validates the env state (i.e. checks whether Init- cmdlet was ran / server requirements are met)

File: cmdlets/sqlserver/New-MiLink.ps1 [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        Write-Verbose "Get VM info [completed]"
+
+        # TODO:
+        # check if managed instance has <100 DBs
+        # Check if MI has enough storage
+
+        # TODO: If we don't have DBM endpoint, we have to create it
+        # we can either create a new certificate for that purpose, or pick an existing one if it meets certain conditions such as:
+        # - "select pvt_key_encryption_type from sys.certificates where name = N'$boxCertName'" must be MK
+        # TODO: can we trust sp_get_endpoint_certificate ?
```
Comment (bruzel) [inline cmdlets/sqlserver/New-MiLink.ps1:163]: No reason it shouldn't work. Documentation explicitly mentions it in relationship with Managed Instance Link.

https://docs.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sp-get-endpoint-certificate-transact-sql?view=sql-server-ver16

Comment (ostojicb) [reply cmdlets/sqlserver/New-MiLink.ps1:163]: I wasn't sure whether the certificate returned by this SP is guaranteed to have its private key encrypted by MK (when ran in on-prem, not MI). i'll try to test for such case and see if it complains

**Pull request merged**